contracts/base/LenderInfo.sol

```diff
@@ -0,0 +1,216 @@
+pragma solidity 0.5.17;
+
+// Libraries
+import "../util/AddressLib.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+// Commons
+import "../util/ZeroCollateralCommon.sol";
+
+// Interfaces
+import "../interfaces/DAIPoolInterface.sol";
+import "../interfaces/LenderInfoInterface.sol";
+import "../interfaces/ZDAIInterface.sol";
+
+contract LenderInfo is LenderInfoInterface {
+    using AddressLib for address;
+    using SafeMath for uint256;
+
+    /* State Variables */
+
+    // last block number of accrued interest
+    uint256 public blockAccruedInterest;
+
+    DAIPoolInterface public daiPool;
+
+    ZDAIInterface public zdai;
+
+    // array of all lending accounts
+    mapping (address => ZeroCollateralCommon.LendAccount) public lenderAccounts;
+
+    /** Modifiers */
+
+    modifier isZDai(address anAddress) {
+        require(areAddressesEqual(address(zdai), anAddress), "Address has no permissions.");
+        _;
+    }
+
+    modifier isDaiPool(address anAddress) {
+        require(areAddressesEqual(address(daiPool), anAddress), "Address has no permissions.");
+        _;
+    }
+
+    modifier isValid(address anAddress) {
+        anAddress.requireNotEmpty("Address is required.");
+        _;
+    }
+
+    /* Constructor */
+
+    constructor(
+        address zdaiAddress,
+        address daiPoolAddress
+    )
+    public
+    {
+        require(zdaiAddress != address(0x0), "ZDai address is required.");
+        require(daiPoolAddress != address(0x0), "Dai pool address is required.");
+        zdai = ZDAIInterface(zdaiAddress);
+        daiPool = DAIPoolInterface(daiPoolAddress);
+        blockAccruedInterest = block.number;
+    }
+
+    /** External Functions */
+
+    function zDaiTransfer(address sender, address recipient, uint256)
+        external
+        isZDai(msg.sender)
+        isValid(sender)
+        isValid(recipient)
+        returns (bool)
+    {
+        // Updating accrued interest for zDAI sender.
+        updateAccruedInterestFor(sender);
+        
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function zDaiMinted(address recipient, uint256)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (bool)
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function zDaiBurnt(address recipient, uint256)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (bool)
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function withdrawInterest(address recipient, uint256 amount)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (uint256)
+    {
+        // Updating accrued interest for zDAI recipient.
```

Comment:
`zDAI recipient` -> `recipient`. They aren't receiving ZDAI I believe? I think they're receiving DAI?

Reply:
Yes, you're right